### PR TITLE
Add header files to ildasm, ilasm vcxproj

### PR DESCRIPTION
**Pull request by 4creators**
Work toward #14884

### CODE REVIEW

**Comment by 4creators** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test

**Comment by 4creators** [comment]
@dotnet-bot test Windows_NT x64 full_opt ryujit CoreCLR Perf Tests Correctness
@dotnet-bot test Windows_NT x64 min_opt ryujit CoreCLR Perf Tests Correctness
@dotnet-bot test Windows_NT x86 full_opt ryujit CoreCLR Perf Tests Correctness
@dotnet-bot test Windows_NT x86 min_opt ryujit CoreCLR Perf Tests Correctness

**Comment by 4creators** [comment]
@dotnet-bot test Ubuntu x64 Checked CoreFX Tests 
@dotnet-bot test Ubuntu x64 Checked Innerloop Build and Test

**Comment by 4creators** [comment]
test Ubuntu arm Cross Checked Innerloop Build and Test
test Ubuntu arm Cross Checked no_tiered_compilation_innerloop Build and Test
test Ubuntu x64 Checked CoreFX Tests

**Comment by 4creators** [comment]
@dotnet-bot test Windows_NT x86 Release Innerloop Build and Test

File: src/ilasm/CMakeLists.txt [hunk 1]
```diff
@@ -25,6 +25,21 @@ set(ILASM_SOURCES
   prebuilt/asmparse.cpp
 )
 if(WIN32)
+  set(ILASM_HEADERS
```
Comment (janvorli) [inline src/ilasm/CMakeLists.txt]: A nit - to make it consistent with other similar PRs of yours, can you please set the ILASM_HEADERS and ILDASM_HEADERS unconditionally and only make adding them to the ILASM_SOURCES conditional?

**Comment by 4creators** [comment]
@janvorli Addressed PR feedback. I don't expect that all tests will pass due to known CI problems.

File: src/ildasm/exe/CMakeLists.txt [hunk 2]
```diff
@@ -35,6 +35,20 @@ set(ILDASM_SOURCES
     ../windasm.cpp
 )
 
+if (WIN32)
```
Comment (janvorli) [inline src/ildasm/exe/CMakeLists.txt]: I've mentioned this one too in my previous comment, but I guess I should have added a separate comment.

Comment (4creators) [reply src/ildasm/exe/CMakeLists.txt]: Fixing

**Comment by 4creators** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test

**Comment by 4creators** [comment]
@janvorli @jkotas IMHO it is good to merge. Failures are expected due to known CI problems.

**Pull request merged**